server/src/main/java/org/opensearch/index/codec/fuzzy/MurmurHash64.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.codec.fuzzy;
+
+import org.apache.lucene.util.BitUtil;
+import org.apache.lucene.util.BytesRef;
+
+/**
+ * Utility to calculate hash for incoming bytes.
+ */
+public class MurmurHash64 {
+    private static final long M64 = 0xc6a4a7935bd1e995L;
+    private static final int R64 = 47;
+    public static final MurmurHash64 INSTANCE = new MurmurHash64();
+
+    /**
+     * Generates a 64-bit hash from byte array of the given length and seed.
+     *
+     * @param data The input byte array
+     * @param seed The initial seed value
+     * @param length The length of the array
+     * @return The 64-bit hash of the given array
+     */
+    public static long hash64(byte[] data, int seed, int offset, int length) {
```

Comment:
If you plan to keep this implementation, please include "self checks" for the hash function to ensure they meet their specs. Take a look at [this example](https://github.com/ketanv3/OpenSearch/blob/58d3394a70622365902d55e500d64dbe6523fef0/libs/common/src/test/java/org/opensearch/common/hash/T1Ha1Tests.java#L16).